cp23/hpi-fun2/motivation/executables/generator.py

```diff
@@ -0,0 +1,232 @@
+from pathlib import Path
+import random
+
+INPUT_LIMIT = 100_000
+
+SAMPLES = [
+["""11 20
+2 5 3 7 2 1 10 4 1 8 2
+1 2 2 2 5 5 1 8 8 8""", "Basic test case; no unreachable nodes."],
+["""5 7
+1 5 10 -2 -2
+1 1 3 3""", "Basic tree case; using negative weights, exceeding weight subtree."],
+["""5 7
+8 5 10 -2 -2
+1 1 3 3""", "No reachable children, bottleneck at root; single coach needed."]
+]
+
+sample = 1
+for test_case, description in SAMPLES:
+    filename = f"sample{sample}"
+
+    Path(filename + ".in").write_text(test_case)
+    Path(filename + ".desc").write_text(description)
+
+    sample += 1
+
+
+def single_parent_single_child():
+    filename = "single_parent_single_child"
+
+    test_case = (
+        "2 20\n"
+        "10 10\n"
+        "1"
+    )
+    Path(filename + ".in").write_text(test_case)
+    Path(filename + ".desc").write_text("Two nodes, with a single child, each being reachable. The child is reachable, but is directly on the barrier.")
+
+
+def negative_root_node_weight():
+    filename = "negative_root_node_weight"
+
+    test_case = (
+        "5 10\n"
+        "-100 90 100 110 120\n"
+        "1 1 1 1"
+    )
+    Path(filename + ".in").write_text(test_case)
+    Path(filename + ".desc").write_text("Root has negative weight and children are partially reachable.")
+
+
+def binary_tree_two_layers():
+    filename = "binary_tree_two_layers"
+
+    test_case = (
+        "7 10\n"
+        "5 5 5 1 1 1 1\n"
+        "1 1 3 3 2 2"
+    )
+    Path(filename + ".in").write_text(test_case)
+    Path(filename + ".desc").write_text("Binary tree of 3 layers with unreachable third layer.")
+
+
+def unreachable_subtree_with_negatives():
+    filename = "unreachable_subtree_with_negatives"
+
+    test_case = (
+        "11 20\n"
+        "1 3 7 -2 2 -6 1 30 3 5 5\n"
+        "1 1 1 3 3 3 4 4 8 8"
+    )
+    Path(filename + ".in").write_text(test_case)
+    Path(filename + ".desc").write_text("Unreachable subtree in a tree including negative weights.")
+
+
+def exponential_weights():
+    filename = "exponential_weights"
+
+    test_case = (
+        "14 15\n"
+        "1 2 2 2 4 4 4 4 8 8 8 8 16 16\n"
+        "1 1 1 3 3 4 4 6 6 8 8 10 10"
+    )
+    Path(filename + ".in").write_text(test_case)
+    Path(filename + ".desc").write_text("Exponentially increasing weights of simple tree.")
+
+
+def single_node_bottleneck():
+    filename = "single_node_bottleneck"
+
+    test_case = (
+        "6 5\n"
+        "1 10 -8 -8 -8 -8\n"
+        "1 2 2 2 2"
+    )
+    Path(filename + ".in").write_text(test_case)
+    Path(filename + ".desc").write_text("There is a single bottleneck in the tree, which renders all its children unreachable.")
+
+
+def large_increasing_chain():
```

Comment:
```suggestion
def large_increasing_chain():
```
Consider replacing chain with path since it is the graph theoretic term.

Reply:
Done 👍🏼